### PR TITLE
[release-3.11] Run nsx role on the first master only

### PR DESCRIPTION
**Pull request by yasensim**
NOTICE
======

Master branch is closed! A major refactor is ongoing in devel-40.
Changes for 3.x should be made directly to the latest release branch they're
relevant to and backported from there.

### CODE REVIEW

File: roles/nsx/tasks/main.yml [hunk 1]
```diff
@@ -1,159 +1,140 @@
...
-  run_once: true
-  import_role:
-    name: kube_proxy_and_dns
-
-- name: NSX | Ensure nsx-system project exists
-  oc_project:
-    name: nsx-system
-    state: present
-    node_selector:
+- when: inventory_hostname == groups.oo_first_master.0
```
Comment (vrutkovs) [inline roles/nsx/tasks/main.yml]: It would be best to rewrite playbooks/openshift-master/private/config.yml to run this role on `oo_first_master` instead

Comment (yasensim) [reply roles/nsx/tasks/main.yml]: it makes sense but if I move it in a separate play in order to provide host: oo_first_master I am losing the restart master handler. I will just add inventory_hostname == groups.oo_first_master.0 in the when clause in playbooks/openshift-master/private/config.yml as oo_first_master is undefined by its own as a var.

Comment (vrutkovs) [reply roles/nsx/tasks/main.yml]: I see, okay, makes sense

File: playbooks/openshift-master/private/config.yml [hunk 2]
```diff
@@ -82,7 +82,7 @@
   - role: calico
     when: openshift_use_calico | default(false) | bool
   - role: nsx
-    when: openshift_use_nsx | default(false) | bool
+    when: (openshift_use_nsx | default(false) | bool) and inventory_hostname == groups.oo_first_master.0
```
Comment (vrutkovs) [inline playbooks/openshift-master/private/config.yml]: For readability
```suggestion
    when:
    - openshift_use_nsx | default(false) | bool
    - inventory_hostname == groups.oo_first_master.0
```

**Comment by vrutkovs** [comment]
/lgtm cancel

Looks good, please squash the commits

**Comment by yasensim** [comment]
@vrutkovs  done it is squashed

**Pull request merged**